### PR TITLE
directory selectFilteredOnly

### PR DESCRIPTION
**Pull request by omrihecht**
https://app.asana.com/0/1159956917255868/1174481783099277/f

### CODE REVIEW

File: projects/ui-framework/bob-table/src/table/table.stories.ts [hunk 1]
```diff
@@ -57,7 +59,7 @@ const template = `
   [shouldAutoSizeColumns]="shouldAutoSizeColumns"
   (rowClicked)="rowClicked($event)"
   (cellClicked)="cellClicked($event)"
-  (selectionChanged)="selectionChanged($event)"
+  (selectionChanged)="selectionChanged($event); onSelectionChanged($event)"
```
Comment (doroncy) [inline projects/ui-framework/bob-table/src/table/table.stories.ts:62]: whats this ?

Comment (omrihecht) [reply projects/ui-framework/bob-table/src/table/table.stories.ts:62]: i added search + filter in the story of b-table, it's for that
image

File: projects/ui-framework/bob-table/src/table/table.stories.ts [hunk 2]
```diff
@@ -217,6 +219,11 @@ function tableStoryFactory({
     sortChanged: action('Sort changed'),
     columnsOrderChanged: action('Column order changed'),
     columnRemoved: action('Column remove button clicked'),
+    totalSelected: 0,
+    onSearchChange: (str, table) => table.gridOptions.api.setQuickFilter(str),
+    onSelectionChanged: function($event) {
+      this.totalSelected = $event.length;
```
Comment (galloween) [inline projects/ui-framework/bob-table/src/table/table.stories.ts:225]: cool! `this` works! 
you could also do it in the onSearchChange method

File: projects/ui-framework/bob-table/src/table/table.stories.ts [hunk 3]
```diff
@@ -47,7 +47,9 @@ const story = storiesOf(ComponentGroupType.Tables, module).addDecorator(
 );
 
 const template = `
-<b-table
+<b-search style="display: block; width: 250px; margin-bottom: 8px;" (searchChange)="onSearchChange($event, table)"></b-search>
+<div style="margin-bottom: 16px;">total selected: {{totalSelected}}</div>
+<b-table #table
```
Comment (galloween) [inline projects/ui-framework/bob-table/src/table/table.stories.ts:52]: this also goes to Notes - check if it looks ok? sometimes if there is too much not related stuff, i do a separate 'template', just for the notes (without the extras)...

File: projects/ui-framework/bob-table/src/table/table.stories.ts (hunk 2 above)
Comment (omrihecht) [reply projects/ui-framework/bob-table/src/table/table.stories.ts:225]: i need to update the selected number in the selectChange event

Comment (galloween) [reply projects/ui-framework/bob-table/src/table/table.stories.ts:225]: ah, 'selected' is not same as 'search results length'

Comment (omrihecht) [reply projects/ui-framework/bob-table/src/table/table.stories.ts:225]: Btw, this works because i used function instead of arrow function

**Pull request merged**